### ISSUE TITLE
arcgis harvesting failed with path in the URL and encoding error

### ISSUE THREAD

**Issue by FuhuXia** [GSA/data.gov#4065]
When arcgis source url contains path component, the job always fails with error message
```
Unable to get content for URL
```

## How to reproduce

Harvest https://gisportal.ibwc.gov/agsportal/

## Expected behavior

successful harvesting.

## Actual behavior
The path `agsportal/` is dropped in the server url when harvester is accessing the source server.
```
Unable to get content for URL: https://gisportal.ibwc.gov/sharing/search?f=pjson&q=modified:[000000000000000000+TO+999999999999999999]&num=100&start=0: HTTPError('404 Client Error: Not Found for url: https://gisportal.ibwc.gov/sharing/search?f=pjson&q=modified:%5B000000000000000000+TO+999999999999999999%5D&num=100&start=0')
```

## Sketch

[Notes or a checklist reflecting our understanding of the selected approach]

**Comment by jbrown-xentity** [comment]
This is our code, so we should be able to figure out what's going on... https://github.com/GSA/ckanext-geodatagov/blob/main/ckanext/geodatagov/harvesters/arcgis.py#L125-L158

**Comment by Jin-Sun-tts** [comment]
the initial issue : urllib.parse.urljoin remove the last item if the source_url does not have '/' at the end, so the solution is: appended '/' to the source URL 

After append the '/', the path shows up correctly. But got encoding error:

[ckanext.harvest.model] Unable to get content for URL: https://gisportal.ibwc.gov/agsportal/sharing/search?f=pjson&q=modified:[000000000000000000+TO+999999999999999999]&num=100&start=301: HTTPError('500 Server Error: Internal Server Error for url: https://gisportal.ibwc.gov/agsportal/sharing/search?f=pjson&q=modified:[000000000000000000+TO+999999999999999999]&num=100&start=301')

looks like it does not encode the '[' and ']' from browser, so manually encode the special characters in the query, but still got error from harvest process like below:

[ckanext.harvest.model] Unable to get content for URL: https://gisportal.ibwc.gov/agsportal/sharing/search?f=pjson&q=modified:%5B000000000000000000%20TO%20999999999999999999%5D&num=100&start=301: HTTPError('500 Server Error: Internal Server Error for url: https://gisportal.ibwc.gov/agsportal/sharing/search?f=pjson&q=modified:%5B000000000000000000%20TO%20999999999999999999%5D&num=100&start=301')

**Comment by jbrown-xentity** [comment]
Strange, https://gisportal.ibwc.gov/agsportal/sharing/search?f=pjson&q=modified:%5B000000000000000000+TO+999999999999999999%5D&num=100&start=301 works in the browser for me. Is this failing locally, or in cloud.gov?
Also, in reviewing the metadata I see all of it is in a variety of languages, none of it english. Anyone know where this source has come from?
![image](https://user-images.githubusercontent.com/26980513/205769635-767c04d6-65c4-4aaf-b3e8-3c001d85152f.png)

**Comment by Jin-Sun-tts** [comment]
The link works in the browser, but failed both at locally and catalog development (cloud.gov)

**Comment by Jin-Sun-tts** [comment]
As there are 100k dataset without extra search criteria, then continue querying those dataset remotely causes errors. 

So we add 'owner:ibwcgis' as the extra search criteria, it limits the return dataset to 39, then it works for this issue.


We got another error related the encoding which does not needed in python3.  

   TypeError: cannot use a string pattern on a bytes-like object

So removed this encode for the title and testing local and on development now.

**Comment by Jin-Sun-tts** [comment]
solved above error but still have other errors return, open new ticket #4108 for different errors.

**Comment by nickumia-reisys** [comment]
This was a crazy solution.  The fact that url encoding was the issue and the only way this was solved was because of this funny story:

Fuhu and Jin were testing the harvesting source url in Chrome for a "long time".  I decided to pair with them and Fuhu copied and pasted the url in Slack for me to open.  When I opened it, it was working.  Everyone was confused, but I was like "I fixed it!" haha  We figured out a few minutes later that Firefox (my daily browser as of writing still) was nice enough to encode some unfriendly characters to send to the arcgis server.  Chrome was lazy and thought the server would be smart.  It wasn't.

Moral of the story (as of 02.02.2023): Firefox is a better browser than Chrome 😉
